server/src/data-layer/services/BookingSlotsService.ts

```diff
@@ -0,0 +1,43 @@
+import FirestoreCollections from "data-layer/adapters/FirestoreCollections"
+import { BookingSlot } from "data-layer/models/firebase"
+import { Timestamp } from "firebase-admin/firestore"
+
+export default class BookingSlotService {
+  // Create
+  public async createBookingSlot(bookingSlotData: BookingSlot) {
+    return await FirestoreCollections.bookingSlots.add(bookingSlotData)
```

Comment:
your tests are currently failing because when you create a document this way it gets assigned a random id, while you try to get it with `TEST_SPID`.

To fix this you have 2 choices:

```ts
  public async createBookingSlot(
    stripeId: string,
    bookingSlotData: BookingSlot
  ) {
    return await FirestoreCollections.bookingSlots
      .doc(stripeId)
      .set(bookingSlotData)
  }
```
Set the id in this function

**OR**

use the return value of this function in the tests

```ts
    const { id } = await bookingSlotService.createBookingSlot(bookingSlotData)
```

Reply:
actually nvm you seem to be getting it by stripe id anyway

Reply:
![image](https://github.com/UoaWDCC/uasc-web/assets/100653148/6bb6e196-f193-4229-a668-67ab3c17fd2f)

If you're talking about delete booking slot it passes because 
```ts
const data = await bookingSlotService.getBookingStripeProductId(TEST_SPID)
```
returns undefined no matter what rn

Reply:
so yea you should choose one of the options, TLDR #1 will mean you can keep using `TEST_SPID` and option #2 means you'll have to update most your tests

Reply:
Okay thanks Benson.